internal/pkg/agent/vault/vault_windows.go

```diff
@@ -59,42 +65,90 @@ func New(path string, opts ...OptionFunc) (v *Vault, err error) {
 		}
 	}
 
-	entropy, err := getOrCreateSeed(path, options.readonly)
+	r := &Vault{
+		path:       path,
+		retryDelay: options.retryDelay,
+		lock:       flock.New(filepath.Join(path, lockFile)),
+	}
+
+	err = r.tryLock(ctx)
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		err = r.unlock(err)
+	}()
+
+	r.entropy, err = getOrCreateSeed(path, options.readonly)
 	if err != nil {
 		return nil, err
 	}
 
-	return &Vault{
-		path:    path,
-		entropy: entropy,
-	}, nil
+	return r, nil
+}
+
+// try to acquire exclusive lock
+func (v *Vault) tryLock(ctx context.Context) error {
+	_, err := v.lock.TryLockContext(ctx, v.retryDelay)
+	if err != nil {
+		return fmt.Errorf("failed to acquire exclusive lock: %v, err: %w", v.lock.Path(), err)
+	}
+	return err
+}
+
+// try to acquire shared lock
+func (v *Vault) tryRLock(ctx context.Context) error {
+	_, err := v.lock.TryRLockContext(ctx, v.retryDelay)
+	if err != nil {
+		return fmt.Errorf("failed to acquire shared lock: %v, err: %w", v.lock.Path(), err)
+	}
+	return err
+}
+
+// unlock unlocks the file lock and preserves the original error if there was a error
+func (v *Vault) unlock(err error) error {
+	unerr := v.lock.Unlock()
+	if err != nil {
+		return err
+	}
+	return unerr
```

Comment:
[Blocker]
Same as in the other one. Please return both errors, you can use `errors.Join` for that.